fix(app): handle detaching 96 when attaching right

sfoster1 opened the pull request:
When we're switching pipettes from a preprotocol flow, we were not
properly handling the case where a 96 is attached and we want to take it
off and attach a pipette to the right mount. Add handling for that.

I'm not really sure how this survived so long because it sure seems like it's always been there.

When reviewing, go commit by commit to see the flow of refactors. Also the logic table looks awful in the diff view unless you make your screen really wide but it's fine if you're just looking at the file 

## testing
- [x] if you have a 96 channel attached, and you set up a protocol with a multi or single on the right mount, does the flow for detaching the 96 and attaching the right mount pipette go correctly

Closes RQA-3123

sfoster1 committed:
refactor(app): pipettewizard: separate logic

Get the steps payload separate from the logic that decides which
payload to return so we have a human hope of comprehending said logic

sfoster1 committed:
refactor(app): pipettewizard: add a lovely table

This table illustrates the reason behind the issue to solve in the next
commit, renames a function since it had a bad name, and will serve as
docs for later use.

sfoster1 committed:
fix(app): handle detaching 96 when attaching right

When we're switching pipettes from a preprotocol flow, we were not
properly handling the case where a 96 is attached and we want to take it
off and attach a pipette to the right mount. Add handling for that.

Closes RQA-3123

caila-marashaj reviewed (approved):
Looks good!

mjhuff reviewed (approved):
Ok, looking through how this interacts with the flows as a whole, I _think_ we're ok.

> not really sure how this survived so long because it sure seems like it's always been there.

Yeah, that surprises me too. 

Tagging @smb2268 for her thoughts if any, since she's most familar with 96 flows.

sfoster1 committed:
fix(api): move axes separately in movetomaint

I don't know why this would stop working now, but something we did makes
the right mount move faster than the left mount when we're moving to the
maintenance position to detach the 96 mount plate, so it trips over its
own feet.

sfoster1 commented:
> Ok, looking through how this interacts with the flows as a whole, I _think_ we're ok.
> 
> > not really sure how this survived so long because it sure seems like it's always been there.
> 
> Yeah, that surprises me too.
> 
> Tagging @smb2268 for her thoughts if any, since she's most familar with 96 flows.

the only change should be (and is based on tests) that the case where there's a 96 attached and you want to have a single-mount, you get the detach-96-first flow instead of the from-empty-mount flow
